docs/examples/NNIG_Hierarchy_NGG.py

```diff
@@ -0,0 +1,268 @@
+
+    Returns
+    -------
+    :obj:`list` of :obj:`float`
+        posterior hyperparameters
+    """
+    data_sum = sum_stats[0]
+    data_sum_squares = sum_stats[1]
+    mu0 = hypers[0]
+    lambda0 = hypers[1]
+    alpha0 = hypers[2]
+    beta0 = hypers[3]
+    if card == 0:
+        return hypers
+    post_hypers = [0] * len(hypers)
+    y_bar = data_sum / card
+    sstat = data_sum_squares - card * (y_bar ** 2)
+    post_hypers[0] = (lambda0 * mu0 + data_sum) / (lambda0 + card)  # mean
+    post_hypers[1] = lambda0 + card  # var_scaling
+    post_hypers[2] = alpha0 + 0.5 * card  # shape
+    num = 0.5 * lambda0 * card * ((y_bar - mu0) ** 2)
+    denom = card + lambda0
+    post_hypers[3] = beta0 + 0.5 * sstat + num / denom  # scale
+    return post_hypers
+
+
+def update_summary_statistics(x, add, sum_stats, state, cluster_data_values):
+    """ Updates cluster statistics when a datum is added or removed from it,
+    the statistics appears in the sampling of the full conditionals of the
+    model for non-conjugate hierarchies and in the computation of the posterior
+    hyperparameters for conjugate hierarchies. In this model, the summary statistics
+    are the current and proposed values of the sum of absolute differences.
+
+    Parameters
+    ----------
+    x : :obj:`list` of :obj:`float`
+        datum (univariate)
+    add : bool
+        if True, the datum has to be added to the cluster, if False, it has to be removed from the cluster
+    sum_stats : :obj:`list` of :obj:`float`
+        list of summary statistics used
+    state : :obj:`list` of :obj:`float`
+        model parameters
+    cluster_data_values : :obj:`list` of :obj:`float`
+        data in the current cluster
+
+    Returns
+    -------
+    :obj:`list` of :obj:`list` of :obj:`float`
+        updated summary statistics and cluster data values
+    """
+    if not len(sum_stats):
+        sum_stats = [0, 0]
+    data_sum = sum_stats[0]
+    data_sum_squares = sum_stats[1]
+    if add:
+        data_sum += x[0]
+        data_sum_squares += x[0] ** 2
+    else:
+        data_sum -= x[0]
+        data_sum_squares -= x[0] ** 2
+    sum_stats = [data_sum, data_sum_squares]
+    return [sum_stats, cluster_data_values]
+
+
+def update_hypers(state, hypers, rng):
+    """Updates the hyperparameters according to the NGG prior assumption:
+    mu0 ~ Normal(mu00, sigma00)
+    lambda0 ~ Gamma(alpha00, beta00)
+    beta0 ~ Gamma(a00, b00)
+    alpha0 is a fixed value
+
+    Parameters
+    ----------
+    state : :obj:`list` of :obj:`float`
+        model parameters
+    hypers : :obj:`list` of :obj:`float`
+        model hyperparameters
+    rng : numpy.random._generator.Generator
+        random number generator to be used when sampling
+
+    Returns
+    -------
+    :obj:`list` of :obj:`float`
+        updated hyperparameters
+    """
+
+    mu00 = 1
+    sig200 = 2.25
+    alpha00 = 0.2
+    beta00 = 0.6
+    a00 = 4.0
+    b00 = 2.0
```

Comment:
you're hardcoding the hyperparameters twice in two different functions. You could have global variables defined just after the imports such as

```
PRIOR_MEAN = 1.0
```

also try to use better naming